### PR TITLE
Attempt to fix issue #3059

### PR DESCRIPTION
**Pull request by jannejava**
If no image is uploaded the setting’s value become null and overwrites the previous value.

### CODE REVIEW

File: src/Http/Controllers/VoyagerSettingsController.php [hunk 1]
```diff
@@ -84,18 +84,24 @@ public function update(Request $request)
...
+                continue;
+            }
+
             $content = $this->getContentBasedOnType($request, 'settings', (object) [
                 'type'    => $setting->type,
-                'field'   => str_replace('.', '_', $setting->key),
+                'field'   => $field,
                 'details' => $setting->details,
                 'group'   => $setting->group,
             ]);
```
Comment (emptynick) [inline src/Http/Controllers/VoyagerSettingsController.php]: Move your condition from line 91 to line 101 and use
`if ($setting->type == 'image' && $content == null)`
This makes everything else unnecessary.

**Review by emptynick — CHANGES_REQUESTED** [review]

**Review by jannejava — COMMENTED** [review]
Of course, I see what you mean. Should I do another pull request?

**Comment by emptynick** [comment]
You just can push to your branch, it will update here automatically.

**Comment by jannejava** [comment]
Did the update of the pull request work?

**Comment by fletch3555** [comment]
Yes it did.

Also, please try to avoid committing whitespace changes (new lines, indentation/spacing changes, etc).  All they do is make the diff harder to read, and make StyleCI revert them after this gets merged.  I understand some IDEs do this automatically, but you can just be more selective in what you commit (e.g. you can select specific lines instead of entire files).

**Comment by jannejava** [comment]
Sorry for that! It must be php-cs-fixer, I have it set on PSR-2 but you are maybe using Symfony style? Anyway, I will try to disable it for this project. 

Do you have any idea if this would fix the disappearing images on the settings page? (it’s a little annoying, when the images disappear ;-))

**Comment by emptynick** [comment]
Yep thats working

**Review by emptynick — APPROVED** [review]

**Review by cyrildewit — APPROVED** [review]
Looks good to me!

**Pull request merged**